### ISSUE TITLE
Date Picker does not update on nzFormat changes

### ISSUE THREAD

**Issue by pburgmer** [NG-ZORRO/ng-zorro-antd#7783]
### Reproduction link
[https://stackblitz.com/edit/angular-wc8wts](https://stackblitz.com/edit/angular-wc8wts)

### Steps to reproduce
Change format in UI at runtime

### What is expected?
Date representation of preselected date in date picker should update as soon as format is changed

### What is actually happening?
Update of selected date representation updates only when user interacts with the date picker

| Environment | Info |
|---|---|
| ng-zorro-antd | 14.3.0 |
| Browser | Chrome |

**Comment by pburgmer** [comment]
I will try to provide a pull request

**Comment by Liiiiimo** [comment]
Are there any updates or solutions?

**Comment by pburgmer** [comment]
No, unfortunately no updates or solutions. I have to check if I still have the changes somewhere. Migrating the component inputs to signals would maybe also fix it. Not sure if ng-zorro has any plan to do so.

**Comment by WwwHhhYran** [comment]
@pburgmer @Liiiiimo By reviewing the source code, I think we only need to add the `updateInputValue` function in the following branch.
https://github.com/NG-ZORRO/ng-zorro-antd/blob/e53e54d0e527f1b9b505372f673f00e4b45ede41/components/date-picker/date-picker.component.ts#L746-L748

I'm going to submit a pr to fix this issue, weelcome to do a code review :)
